File: nomad-samples/js/account-authenticaton/account-authentication.js
import login from "./account/login.js";
import logout from "./account/logout.js";
import refreshToken from "./account/refresh-token.js";
import forgotPassword from "./account/forgot-password.js";
import resetPassword from "./account/reset-password.js";

const LOGIN_FORM = document.getElementById("loginForm");
const LOGOUT_FORM = document.getElementById("logoutForm");
const AUTH_FORM = document.getElementById("authForm");
const REFRESH_AUTH_FORM = document.getElementById("refreshAuthForm");
const FORGOT_PASS_FORM = document.getElementById("forgotPassForm");
const RESET_PASS_FORM = document.getElementById("resetPassForm");

const USERNAME_INPUT = document.getElementById("usernameInput");
const PASSWORD_INPUT = document.getElementById("passwordInput");
const APPLICATION_ID_LOGIN_INPUT = document.getElementById("applicationIdLoginInput");
const TOKEN_INPUT = document.getElementById("authInput");
const FORGOT_PASSWORD_USERNAME_INPUT = document.getElementById("forgotPasswordUsernameInput");
const RESET_PASSWORD_INPUT = document.getElementById("resetPasswordInput");
const APPLICATION_ID_LOGOUT_INPUT = document.getElementById("applicationIdLogoutInput");

sessionStorage.clear();

LOGIN_FORM.addEventListener("submit", function (event) 
{
    event.preventDefault();
    let username = USERNAME_INPUT.value;
    let password = PASSWORD_INPUT.value;
    let applicationId = APPLICATION_ID_LOGIN_INPUT.value;
    loginUser(username, password, applicationId);
});

AUTH_FORM.addEventListener("submit", function (event)
{
    event.preventDefault();
    sessionStorage.setItem("token", TOKEN_INPUT.value);
    console.log("Successfuly updated token");
});

REFRESH_AUTH_FORM.addEventListener("submit", function (event)
{
    event.preventDefault();
    refreshTokenMain();
});

FORGOT_PASS_FORM.addEventListener("submit", function (event) 
{
    event.preventDefault();
    let username = FORGOT_PASSWORD_USERNAME_INPUT.value;
    forgotPass(username);
});

RESET_PASS_FORM.addEventListener("submit", function (event) 
{
    event.preventDefault();
    let username = USERNAME_INPUT.value;
    let code = RESET_PASSWORD_INPUT.value;
    resetPass(username, code);
});

LOGOUT_FORM.addEventListener("submit", function (event) 
{
    event.preventDefault();
    let applicationId = APPLICATION_ID_LOGOUT_INPUT.value;
    logoutUser(applicationId);
});

async function loginUser(USERNAME, PASSWORD, APPLICATION_ID)
{
    try
    {
        console.log("Logging in");
        const TOKEN = await login(USERNAME, PASSWORD, APPLICATION_ID);
        console.log("Successfully Logged in");
        console.log(`Token: ${TOKEN}`);
    }
    catch (error)
    {
        throw new Error(error);
    }
}

async function refreshTokenMain()
{
    if (!sessionStorage.getItem("token"))
    {
        throw new Error("Authentication token: The authentication token is empty");
    }

    try
    {
        console.log("Refreshing token");
        await refreshToken();
        console.log(`Token: ${sessionStorage.getItem("token")}`);
    }
    catch (error)
    {
        throw new Error(error)
    }
}

async function forgotPass(USERNAME)
{
    try
    {
        console.log("Sending code");
        await forgotPassword(USERNAME);
        console.log("An email has been sent to you with a 6 digit code");
    }
    catch (error)
    {
        throw new Error(error)
    }
}

async function resetPass(USERNAME, CODE)
{
    try
    {
        console.log("Resetting password");
        await resetPassword(USERNAME, CODE);
        console.log("Password reset successfully");
    }
    catch (error)
    {
        throw new Error(error);
    }
}

async function logoutUser(applicationId)
{
    if (!sessionStorage.getItem("intervalId"))
    {
        throw new Error("Interval id: The interval id is empty");
    }

    try
    {
        if (applicationId = "") applicationId = "00000000-0000-0000-0000-000000000000"

        console.log("Logging out");
        logout(applicationId);
        console.log("Successfully logged out")
    }
    catch (error)
    {
        throw new Error(error);
    }
}

File: nomad-samples/python/account_authentication/account_authentication.py
from account.login import *
from account.refresh_token import *
from account.forgot_password import *
from account.reset_password import *
from account.logout import *
from constants.project_constants import*
from exceptions.api_exception_handler import *

import json

def login_main():
    try:
        while True:
            USERNAME = input("Enter your username: ")
            PASSWORD = input("Enter your password: ")
            APPLICATION_ID = input("Enter your application id (optional): ")
            print("Logging in")
            LOGIN_INFO = login(USERNAME, PASSWORD, APPLICATION_ID)  
            if LOGIN_INFO != "Login info incorrect":
                break
            print("Login credentials are incorrect.")
        
        print(f"Token: {LOGIN_INFO['token']}")
        LOGIN_INFO["username"] = USERNAME
        return LOGIN_INFO, APPLICATION_ID
    except:
        raise Exception()
    
def refresh_token_main(REFRESH_TOKEN):
    try:
        print("Refreshing token")
        REFRESH_INFO = refresh_token(REFRESH_TOKEN)
        print(f"Token: {REFRESH_INFO['token']}")
        token = REFRESH_INFO["token"]
    except:
        raise Exception

def reset_password_main(AUTH_TOKEN, USERNAME):
    try:
        forgot_password(USERNAME, AUTH_TOKEN)

        print("An email has been sent to you with a 6 digit code")
        while True:
            verification_input = input("Did you recieve the 6 digit code (y/n): ")
            if verification_input == "y":
                while True:
                    IS_INT = True
                    CODE = input("Enter 6 digit code: ")
                    try: 
                        int(CODE)
                    except:
                        IS_INT = False
                    if len(CODE) == 6 and IS_INT:
                        try:
                            print("Resetting password")
                            NEW_PASSWORD = input("Enter new password: ")
                            reset_password(USERNAME, CODE, NEW_PASSWORD)
                            print("Password Reset")
                            break
                        except:
                            print("The 6 digit code you have provided is invalid")
                    else:
                        print("The 6 digit code you have provided is invalid")
                break
            else:
                print("Resending 6 digit code")
                forgot_password(USERNAME, AUTH_TOKEN)
                print("An email has been sent to you with a 6 digit code")
    except:
        raise Exception()

if __name__ == "__main__":
    LOGIN_INFO, APPLICATION_ID = login_main()
    while True:
        print("Do you want to refresh your token, reset your password, or logout?")
        USER_INPUT = input("Enter refresh to refresh token, reset to reset your password or logout: ")
        auth_token = LOGIN_INFO["token"]
        if USER_INPUT == "refresh":
            auth_token = refresh_token_main(LOGIN_INFO["refreshToken"])
        elif USER_INPUT == "reset":
            reset_password_main(auth_token, LOGIN_INFO["username"])
        elif USER_INPUT == "logout":
            logout(auth_token, LOGIN_INFO["userSessionId"], APPLICATION_ID)
            print("Logged out successfully")
            break
        else:
            print("The input is incorrect")

